### DISCUSSION TITLE
Mock QueryRaw

### DISCUSSION THREAD

**Discussion by mwarnerdotme** [prisma/prisma-client-go#1333] [Q&A]
Does anyone know how to mock QueryRaw calls? All mocks are attached to specific models and there doesn't seem to be a clean/easy way to mock a query that returns RawJSON from a jsonb column, for example. I tried adding a custom expectation to the mock server's expectations slice but those expectations also seem to be attached to a specific model. This is my custom SQL query in my route handler:

```go
selectChannelConfigsWithStructQuery := fmt.Sprintf(`
SELECT * FROM app.channels
WHERE config->'nodes' @> '[{"metadata": {"dependencies": {"structs": [{"structId": "%s"}]}}}]'::jsonb
LIMIT 1;
`, request.Input.StructId)

var channelConfigsWithStruct []db.RawJSON
err := prisma.Client.Prisma.QueryRaw(selectChannelConfigsWithStructQuery).Exec(context.Background(), &channelConfigsWithStruct)
```

**Comment by steebchen** [comment]
Mocking raw queries is currently not supported unfortunately. It should be possible to add, I just haven't had the feature request until now.

**Reply by mwarnerdotme** [reply]
That's good to know! I can attempt to implement it and open a PR. If you have any pointers, let me know!

I suppose the alternative would be a jsonb column query that would replace the need for a raw query in the first place. I kind of assumed that this wasn't really supported (or performant) by the ORM, but let me know if I'm approaching this raw query with the wrong idea.

**Reply by steebchen** [reply]
you should be able to do some basic json queries, but not everything might be supported. 

this should explain how it works:
https://goprisma.org/docs/reference/features/json 

for all filters you can check https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/working-with-json-fields

(but note not everything is supported in the go client)

**Reply by steebchen** [reply]
and yeah if you do need it and you make a PR I'd be happy to merge it, but it would need a test case as well. 

some links on how the mocks work

test cases: https://github.com/steebchen/prisma-client-go/blob/6e49bf471a7f3685497546a0344152101b706c50/test/mock/mock_test.go

templates to generate mock values in the generated client: https://github.com/steebchen/prisma-client-go/blob/953a390d96563632fa02e0b1023fb3b71214d562/generator/templates/mock.gotpl

**Reply by mwarnerdotme** [reply]
I read the documentation and tried to get my particular JSON object to work for an hour or two. I couldn't seem to get it to work in JS/TS or in Go so I gave up and decided that I'll just query for all the objects that I need and do the JSON filtering on the server side instead of in the DB query.

Thanks for your help though!

**Reply by steebchen** [reply]
yeah if it's complex it might be simpler to just do a query or as you said on the backend if your use case allows it 👍
